src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -662,8 +662,8 @@ public IndexDefinition GetIndexDefinition()
                         result.IndexTimestamp = stats.LastIndexingTime ?? DateTime.MinValue;
                         result.LastQueryTime = stats.LastQueryingTime ?? DateTime.MinValue;
                         result.ResultEtag = CalculateIndexEtag(Definition, result.IsStale,
-                            lastDocEtags: Collections.Select(x => DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(documentsContext, x)),
-                            lastMappedEtags: Collections.Select(x => _indexStorage.ReadLastMappedEtag(indexTx, x)));
+                            lastDocEtags: Collections.Select(x => DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(documentsContext, x)).ToList(),
+                            lastMappedEtags: Collections.Select(x => _indexStorage.ReadLastMappedEtag(indexTx, x)).ToList());
```

Comment:
The ToArray is actually doing a List inside.
I was refering to removing the Select(x=>) lambda because is is causing allocations in favor of foreach